geospaas_processing/cli/download.py

```diff
@@ -0,0 +1,78 @@
+"""
+Download files that are selected from the database using input criteria.
+"""
+import argparse
+import os
+import time
+from datetime import datetime
+import json
+import django
+
+from dateutil.relativedelta import relativedelta
+from dateutil.tz import tzutc
+from django.contrib.gis.geos import GEOSGeometry
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'geospaas_processing.settings')
+django.setup()
+from geospaas.catalog.models import Dataset
+import geospaas_processing.downloaders as downloaders
+
+
+def main(arg):
+    cumulative_query=json.loads(arg.query) if arg.query else {}
+    if arg.rel_time_flag:
+        designated_begin = datetime.now().replace(tzinfo=tzutc()) + relativedelta(
+            hours=-abs(int(arg.begin)))
+        designated_end = datetime.now().replace(tzinfo=tzutc())
+    else:
+        designated_begin = datetime.strptime(arg.begin, "%Y-%m-%d").replace(tzinfo=tzutc())
+        designated_end = datetime.strptime(arg.end, "%Y-%m-%d").replace(tzinfo=tzutc())
```

Comment:
This would be much easier to test if you split this into functions.

Reply:
Done in the next commit.